src/persist/src/indexed/mod.rs

```diff
@@ -132,7 +193,7 @@ impl Pending {
 ///   AppliedState (which has no knowledge of storage, etc). Then, if this was
 ///   successful, Indexed will serialize AppliedState and durably write it down.
 #[derive(Debug)]
-pub struct Indexed<L: Log, B: Blob> {
+pub struct Indexed<L: Log, B: BlobRead> {
```

Comment:
do we think it's worth both of these sharing an Indexed? for example, it seems wrong for the read one to even have maintainer and pending. OTOH maintainer is also going away in the "full" version. since the read only commands mostly just call into state, the only code they really share is registering listeners and (soon) sending them things

Reply:
both of these being Blob and BlobRead, and the alternative being to split out `IndexedRead<L: Log, B: BlobRead>` right? I think since the time this question was posed we have moved towards `IndexedRead`. Let me know if I'm missing something

Reply:
yeah wrote this comment before we had that convo. that's the preferred plan at this point. only thing is this refactor is complicated enough that until i actually prototype it, i'll stay skeptical